### PR TITLE
Added new testcase:Kill NFS executor pod during backup and restore

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes # PA-1671

**Special notes for your reviewer**:
1)Added a retry in DeletePodWithLabelInNamespace as nfs executor pod takes time to come up after backup/restore starts
2)Added retry in ValidateBackup as backup takes time to complete once we delete the nfs executor pod
3)Added helper method DeletePodWhileBackupInProgress and DeletePodWhileRestoreInProgress(
4)Added new testcase: DeleteNfsExecutorPodWhileBackupAndRestoreInProgress
5)Renamed CreateBackupWithCustomResourceType to CreateBackupWithCustomResourceTypeWithoutValidation as this method is to check backup with custom resource without validation.
There already exist a method : CreateBackupWithCustomResourceTypeWithValidation

log: https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/2837/consoleFull

RKE: https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/2849/consoleFull

https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/basic-system-test/job/px/job/NFS/job/PX-Cloud-Vanilla-NFS/177/consoleFull

### CODE REVIEW

File: tests/backup/backup_helper.go [hunk 1]
```diff
@@ -1438,11 +1437,21 @@ func GetAllRestoresNonAdminCtx(ctx context.Context) ([]string, error) {
 
 // DeletePodWithLabelInNamespace kills pod with the given label in the given namespace
 func DeletePodWithLabelInNamespace(namespace string, label map[string]string) error {
-	pods, err := core.Instance().GetPods(namespace, label)
+	var pods *corev1.PodList
+	podList := func() (interface{}, bool, error) {
+		pods, _ = core.Instance().GetPods(namespace, label)
```
Comment (kphalgun-px) [inline tests/backup/backup_helper.go:1442]: Does it return an error?

**Review by kphalgun-px — APPROVED** [review]
LGTM

File: tests/backup/backup_helper.go (hunk 1 above)
Comment (snigdha-px) [reply tests/backup/backup_helper.go]: corrected

File: tests/backup/backup_nfs_test.go [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+				},
+				SchedulePolicy: schedulePolicyInfo,
+			}
+			_, err = Inst().Backup.CreateSchedulePolicy(ctx, userSchedulePolicyCreateRequest)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation schedule policy %s", schedulePolicyName))
+			schedulePolicyUID, err = Inst().Backup.GetSchedulePolicyUid(orgID, ctx, schedulePolicyName)
+			log.FailOnError(err, "failed to fetch uid for schedule policy- %s", schedulePolicyName)
+		})
+
+		Step("Taking schedule backup of multiple namespaces", func() {
```
Comment (kshithijiyer-px) [inline tests/backup/backup_nfs_test.go:105]: Can we move this entire thing into a loop ? Like have a dict and say run for 3 times for all three variations?

**Comment by mkoppal-px** [comment]
Can we get a test run for this PR?

File: tests/backup/backup_helper.go [hunk 3]
```diff
@@ -1438,11 +1437,33 @@ func GetAllRestoresNonAdminCtx(ctx context.Context) ([]string, error) {
 
 // DeletePodWithLabelInNamespace kills pod with the given label in the given namespace
 func DeletePodWithLabelInNamespace(namespace string, label map[string]string) error {
-	pods, err := core.Instance().GetPods(namespace, label)
+	var pods *corev1.PodList
+	var err error
+	podList := func() (interface{}, bool, error) {
+		pods, err = core.Instance().GetPods(namespace, label)
+		log.Infof(" 111 The value of pod is %v", pods.Items)
+		log.Infof(" The value of err is %v", err.Error())
```
Comment (mkoppal-px) [inline tests/backup/backup_helper.go:1445]: Please remove or fix the log messages

File: tests/backup/backup_nfs_test.go (hunk 2 above)
Comment (mkoppal-px) [reply tests/backup/backup_nfs_test.go]: Not sure why we need a loop and switch case here? We can simply split this into 3 different steps right?
One step each for schedule backup, custom resource backup and multi namespace backup.

File: tests/backup/backup_helper.go (hunk 3 above)
Comment (snigdha-px) [reply tests/backup/backup_helper.go]: done

File: tests/backup/backup_nfs_test.go (hunk 2 above)
Comment (snigdha-px) [reply tests/backup/backup_nfs_test.go]: @kshithijiyer-px @mkoppal-px 
Have made the changes as suggested by @kshithijiyer-px and I also feel that the code is more clean and less redundant now.
Initially I had 3 function repeated each for schedule backup, custom resource backup and multi namespace backup.

**Review by mkoppal-px — APPROVED** [review]
LGTM
Please check on RKE and IBM pipelines

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Pull request merged**